library/common-library/templates/_naming.tpl

```diff
@@ -18,13 +28,16 @@ We truncate at 63 chars because some Kubernetes name fields are limited to this
 {{- define "newrelic.common.naming.fullname" -}}
 {{- $name := "" }}
 
-{{- if .Values.fullnameOverride }}
-    {{- $name = .Values.fullnameOverride  }}
-{{- else }}
-    {{- $name = printf "%s-%s" .Release.Name (include "newrelic.common.naming.name" .)}}
-{{- end }}
+{{- if .Values.fullnameOverride -}}
+    {{- $name = .Values.fullnameOverride  -}}
+{{- else -}}
+    {{- $name = include "newrelic.common.naming.name" .  -}}
+    {{- if not ( contains $name .Release.Name ) -}}
+        {{- $name = printf "%s-%s" .Release.Name (include "newrelic.common.naming.name" .)}}
```

Comment:
NIT:

```suggestion
        {{- $name = printf "%s-%s" .Release.Name $name .)}}
```
I am not completely sure if it would work or if it is more idiomatic... WDYT?

Reply:
i guess it should work since that variable is in the scope

Reply:
I copy and pasted and it broke because of that `.`... Sorry for this half-hour debugging something I could not even understand....

Fixed the readability on [#a67f263](https://github.com/newrelic/helm-charts/pull/784/commits/a67f2631c497b4422b42031459a1ff58276eed4b)